src/services/geometry/richgeo/IrtGeoDRICH.cc

```diff
@@ -186,7 +185,44 @@ void richgeo::IrtGeoDRICH::DD4hep_to_IRT() {
   // define the `cell ID -> pixel position` converter
   SetReadoutIDToPositionLambda();
 }
-
+TVector3 richgeo::IrtGeoDRICH::GetSensorSurface(CellIDType id){
+  printf("This is just a test\n");
+  auto nSectors= m_det->constant<int>("DRICH_num_sectors");
+  TVector3 normX(1, 0,  0); // normal vectors
+  TVector3 normY(0, -1, 0);
+  TVector3 sensorNorm;
+  auto cellMask = uint64_t(std::stoull(m_det->constant<std::string>("DRICH_cell_mask")));
+  for (int isec = 0; isec < nSectors; isec++) {
+    std::string secName = "sec" + std::to_string(isec);
+    auto sensorThickness = m_det->constant<double>("DRICH_sensor_thickness") / dd4hep::mm;
+    auto sensorSize      = m_det->constant<double>("DRICH_sensor_size") / dd4hep::mm;
+    for(auto const& [de_name, detSensor] : m_detRich.children()) {
```

Comment:
Looping over sensors is not really practical, and violates DRY. Please make use of `m_sensor_info`:
https://github.com/eic/EICrecon/blob/46f7d9a7fba79a1c2661897c8785a48cea36c3de/src/services/geometry/richgeo/IrtGeo.h#L62

You can add a new member to `richgeo::Sensor` for the surface norm:
https://github.com/eic/EICrecon/blob/46f7d9a7fba79a1c2661897c8785a48cea36c3de/src/services/geometry/richgeo/RichGeo.h#L20-L27

It gets filled here, where you already have `normZ`:
https://github.com/eic/EICrecon/blob/46f7d9a7fba79a1c2661897c8785a48cea36c3de/src/services/geometry/richgeo/IrtGeoDRICH.cc#L137-L146

Reply:
So at `GetSensorSurface(CellIDType)`, I need to pass this `id & cellMask` somehow to get the sensor info. Right?

Reply:
Yes, just use `m_sensor_info` for the lookup.

Reply:
> Yes, just use `m_sensor_info` for the lookup.

is there a method called lookup? Or any example?

Reply:
Ah it was stupid question again. m_sensor_info is a map.
so 
auto my_normal  =  m_senso_info->constant<dd4hep::Direction>("surface_normZdir"), if I insert the normZdir as sensor_normZdir inside Sensor class. Right?

Reply:
The map is 1-to-1 sensor ID to a `richgeo::Sensor` object. Observe the usage of `m_sensor_info` in other places.

`cellID` masked by the cell mask is the sensor ID, if I recall correctly; double check how it is defined in `epic`.

Reply:
> https://github.com/eic/EICrecon/blob/46f7d9a7fba79a1c2661897c8785a48cea36c3de/src/services/geometry/richgeo/RichGeo.h#L20-L27

@c-dilks what is this `int` here? I understood that the map is between sensor id and sensor info. Is it right?

Reply:
> `cellID` masked by the cell mask is the sensor ID, if I recall correctly; double check how it is defined in `epic`.

It is like that, I was using it. `sensorID = cellID & cellmask`

Reply:
So if I am not wrong this `DD4hep_to_IRT` is called independently. I mean I still can call some of the sensor info without creating objects which are defined inside `DD4hep_to_IRT`.

Reply:
> @c-dilks what is this `int` here? I understood that the map is between sensor id and sensor info. Is it right?

Yes, the `int` is the sensor ID, serving as the map key. The map value is the sensor info, a `richgeo::Sensor` object.

Reply:
> So if I am not wrong this `DD4hep_to_IRT` is called independently. I mean I still can call some of the sensor info without creating objects which are defined inside `DD4hep_to_IRT`.

It's called in the constructor: https://github.com/eic/EICrecon/blob/a73bc36547a8f5d81266234b5a17d60d8ace9139/src/services/geometry/richgeo/IrtGeoDRICH.h#L13-L14